### ISSUE TITLE
GetOccupiedBytes tier statistic value is not updated after cache clearing

### ISSUE THREAD

**Issue by karmann-dm** [ehcache/ehcache3#3034]
### Prerequesites
Java 11, Ehcache 3.10.0

### Problem statement

The dedicated statistics value fetched from `getOccupiedBytes` method of `TierStatistics` is not updated after `cache.clear()` call on Ehcache instance. However it is updated correctly after `cache.remove(key)`.

### Example
```
final var statisticsService = new DefaultStatisticsService();
final var cacheManager = CacheManagerBuilder.newCacheManagerBuilder()
                .using(statisticsService)
                .build(true);

final var cacheConfiguration = CacheConfigurationBuilder
                .newCacheConfigurationBuilder(
                        String.class, String.class,
                        ResourcePoolsBuilder
                                .newResourcePoolsBuilder()
                                .heap(100, MemoryUnit.KB))
                .withExpiry(ExpiryPolicyBuilder.timeToLiveExpiration(Duration.ofSeconds(100)))
                .build();

final var cache = cacheManager.createCache("cacheName", cacheConfiguration);

// Fill cache with some dummy values
cache.put("key1", "value1");
cache.put("key2", "value2");
cache.put("key3", "value3");

final var currentMemory = statisticsService.getCacheStatistics("cacheName").getTierStatistics().get("OnHeap").getOccupiedByteSize();

cache.clear();

final var newMemory = statisticsService.getCacheStatistics("cacheName") .getTierStatistics().get("OnHeap").getOccupiedByteSize();

// As a result newMemory is equal to currentMemory, however if we call 
// cache.remove("key1"); cache.remove("key2"); and cache.remove("key3") - getOccupiedByteSize() will return 0 after all these operations
```

**Comment by chrisdennis** [comment]
@nnares could you validate this in-house and label as a `bug` as appropriate.

**Comment by chrisdennis** [comment]
Fixed in #3077
